### PR TITLE
add natural magnitude systems to sncosmo + more support for CSP data

### PR DESCRIPTION
**Pull request by dannygoldstein**
This PR augments the existing `MagSystem` class hierarchy in `sncosmo` by defining a `NaturalMagSystem` class. The purpose of this enhancement is to allow users to define magnitude systems in which the magnitudes / fluxes are calibrated such that they correspond to what one would measure through filters attached to a specific telescope or set of telescopes, rather than what one would measure in some standard set of filters or in a magnitude system based on a _single_ spectrophotometric standard. `NaturalMagSystem` inherits directly from `MagSystem`, and can be instantiated given the zeropoints, bands, and SEDs of the spectrophotometric standards of each band of the natural system.  Public interfaces to existing `MagSystems` (`SpectralMagSystem`, `ABMagSystem`) remain unchanged, but they now inherit from another abstract class, `StandardMagSystem`. This class is essentially what `MagSystem` was before. `MagSystem` has been stripped down somewhat, but its abstract interface has not changed.  

This PR also provides a new format for the `read_lc` method, `format='csp'`, as discussed in #116. 
-  [x] Make `read_lc(format='csp')` conform to the `sncosmo` data model

The CSP natural magnitude system (http://csp.obs.carnegiescience.edu/data/filters) is implemented as a builtin. 

Some unit tests are provided for the `NaturalMagSystem` class using the CSP natural photometric system.

Some of the commits in this PR are also present in #115.

### CODE REVIEW

**Comment by kbarbary** [comment]
With #115 now merged, can you rebase this on master to make it easier to review?

**Comment by dannygoldstein** [comment]
Done.

**Comment by kbarbary** [comment]
Interestingly, I originally included this sort of functionality in sncosmo, calling it a "local" magnitude system. This [this removal commit](https://github.com/sncosmo/sncosmo/commit/0bc63b6a041dceafb4a9f5498de50a6a42e4c762) for what my minimal version looked like before it was removed. Having a separate class for this like you've done here seems like a better idea.

A couple incomplete thoughts:

- Terminology: as far as I know, in the context of magnitude systems, "natural" refers to the bandpass the magnitude is measured in, not the standard it is referenced to. For example, reporting Vega magnitudes in the desg, desr, ... filters from DES measurements would be natural magnitudes and also Vega magnitudes. On the other hand, applying some color corrections and reporting Bessell magnitudes from DES observations would not be natural magnitudes. This is why I chose the terminology "local" magnitude system when trying to implement this myself.

- Some such systems are referenced to AB, such as "SDSS magnitudes" which are AB, but with some fixed small offset in each band. To allow for such systems, it might be better for `NaturalMagSystem` to reference other system instances internally.

File: .gitignore [hunk 1]
```diff
@@ -50,3 +50,4 @@ sncosmo/cython_version.py
 
 # ipython notebooks
 *.ipynb
+.cache/
```
Comment (kbarbary) [inline .gitignore]: What creates `.cache`?

File: sncosmo/spectral.py [hunk 2]
```diff
@@ -468,16 +500,89 @@ def zpbandflux(self, band):
...
+    def bands(self):
+        return self._bands
+    
+    @property
+    def standards(self):
+        return self._standards
+
+    @_verify_band(0)
+    def _refspectrum_bandflux(self, band):
+        return self.standards[band].bandflux(band)
```
Comment (kbarbary) [inline sncosmo/spectral.py]: I think we could forgo the use of a decorator here, and just use a normal method. This is just as compact:

```python
def _refspectrum_bandflux(self, band):
    self._verify_band(band)
    return self.standards[band].bandflux(band)
```
and easier to parse.

File: sncosmo/spectral.py [hunk 3]
```diff
@@ -468,16 +500,89 @@ def zpbandflux(self, band):
...
+    standards: iterable of `~sncosmo.Spectrum`,
+        The spectrophotmetric flux standards for each band.
+    """
+
+    def __init__(self, bands, zeropoints, standards, name=None):
+        super(NaturalMagSystem, self).__init__(name)
+        if len(bands) != len(zeropoints):
+            raise ValueError, 'Number of passed bands (%d) does not equal '  \
+                'the number of passed zeropoints (%d).'  % (len(bands),      
+                                                            len(zeropoints))
```
Comment (kbarbary) [inline sncosmo/spectral.py]: Minor style notes for consistency with rest of code base: (1) Instantiate exceptions with parentheses, (2) don't use `\` for line continuation, (3) use `.format` over `%`. E.g.,
```python
raise ValueError('Number of passed bands ({}) does not equal '
                 'the number of passed zeropoints({}).'
                 .format(len(bands), len(zeropoints)))

**Comment by kbarbary** [comment]
Superceded by #126.

**Pull request closed without merging**